### ISSUE TITLE
Homekit registration is lost each time my router is restarted

### ISSUE THREAD

**Issue by bvial** [nfarina/homebridge#1903]
I use homebrige on raspberrypi and since i'm update homebrige from 0.4.41 to 0.4.44.
each time the router of my local network is restart , homebrige mdns registration is lost.
i use  dns-sd -B _hap._tcp to check registration. 
I need to restart homebrige to see back the registration
i guess the issue is come from the change done in HAP-NodeJS with the replace of mdns by bonjour-hap

**Comment by NorthernMan54** [comment]
When your router restarts does your raspberry pi change IP address or does it stay the same?

**Comment by bvial** [comment]
I declare static IP addresses for my raspberry pi on Routers, so the ip didn't change. 
Every thing was working well before i update nodejs ( 8 to 10 ) and homebrige (0.4.41 to 0.4.44.).
on OS X, before router reboot , with dns-sd -B _hap._tcp 
$  dns-sd -B _hap._tcp
Browsing for _hap._tcp
DATE: ---Sat 23 Jun 2018---
19:23:04.368  ...STARTING...
Timestamp     A/R    Flags  if Domain               Service Type         Instance Name
19:23:04.369  Add        3  12 local.               _hap._tcp.           Philips hue - 22509D
19:23:04.370  Add        2  12 local.               _hap._tcp.           Homebridge-CAD8

After i reboot my router Homebridge-CAD8 is not more persent,
$dns-sd -B _hap._tcp
Browsing for _hap._tcp
DATE: ---Sat 23 Jun 2018---
19:31:56.123  ...STARTING...
Timestamp     A/R    Flags  if Domain               Service Type         Instance Name
19:31:56.123  Add        2  12 local.               _hap._tcp.           Philips hue - 22509D

I restart Homebridge on raspberry pi and now Homebridge-CAD8 is back
 dns-sd -B _hap._tcp
Browsing for _hap._tcp
DATE: ---Sat 23 Jun 2018---
19:36:28.364  ...STARTING...
Timestamp     A/R    Flags  if Domain               Service Type         Instance Name
19:36:28.365  Add        3  12 local.               _hap._tcp.           Philips hue - 22509D
19:36:28.366  Add        2  12 local.               _hap._tcp.           Homebridge-CAD8

**Comment by michaeljanich** [comment]
I have the same issue. I traced it to be a Raspberry Pi Linux Wifi problem: the wifi connection is NOT reconnecting, after the router has restarted.

In my home, when the router restarts, I have to restart all 10 Raspberry PIs afterwards (except for the one, that uses a wired network).

I googled a bit and only found some code, that does a ping and if it fails, resets the wifi (ifconfig wlan0 down; sleep 3; ifconfig wlan0 up). Is that really the **best** solution?

**Comment by bvial** [comment]
thanks for you  message,  but my Raspberry  uses a wired network.

**Comment by Kydix** [comment]
Same for me ! 

My raspberry is connected with cable, and has a static IP. Each time i reboot my router, all my accessories are unreachable until i reboot homebridge

**Comment by ebaauw** [comment]
You might want to disable IGMP snooping on your router.

**Comment by Kydix** [comment]
Hum... i'm in france and i use the box provided by my ISP. I don't think i can disable IGMP snooping in the settings

**Comment by fabsfuchs** [comment]
Hello, I've got the same problem. After every restart of my router I have to restart the homekit core to work again. I have two pi's one wired and one wireless. The problem is on both the same. 
After the reconnect of the pi I started a portscan to check if the homekit port (51826) is open or closed. Its open and I can also see the portscanner request in the debug log.
Then I tried a mdns scanner on my iPhone. Here I cant see the service. Maybe it has something to do with mdns???
